src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsLegacyWebUi/WindowsFormsWebAuthenticationDialogBase.cs

```diff
@@ -117,24 +117,33 @@ protected virtual void WebBrowserNavigatingHandler(object sender, WebBrowserNavi
                 e.Cancel = true;
             }
 
+            if (string.IsNullOrEmpty(e.Url))
```

Comment:
I think `e.Url` will always be a valid URL here but adding a check just in case. In `WebBrowserNavigatingEventArgs` `Url` is actually of type `Uri`. Since we get URL as string from `BeforeNavigate2`, thought it would be better to do a check, log, and cast here than in the event args.